scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/models/SchedulingDSL.java

```diff
@@ -109,20 +120,31 @@ private static final JsonObjectParser<GoalSpecifier.CoexistenceGoalDefinition> c
             .optionalField("activityFinder", activityExpressionP)
             .field("alias", stringP)
             .field("forEach", constraintExpressionP)
-            .optionalField("startConstraint", activityTimingConstraintP)
-            .optionalField("endConstraint", activityTimingConstraintP)
+            .optionalField("startConstraint", (JsonParser<? extends TimingConstraint>) chooseP(activityTimingConstraintP, activityTimingConstraintFlexibleRangeP))
+            .optionalField("endConstraint", (JsonParser<? extends TimingConstraint>) chooseP(activityTimingConstraintP, activityTimingConstraintFlexibleRangeP))
             .field("shouldRollbackIfUnsatisfied", boolP)
-            .map(
-                untuple(GoalSpecifier.CoexistenceGoalDefinition::new),
-                goalDefinition -> tuple(
-                    goalDefinition.activityTemplate(),
-                    goalDefinition.activityFinder(),
-                    goalDefinition.alias(),
-                    goalDefinition.forEach(),
-                    goalDefinition.startConstraint(),
-                    goalDefinition.endConstraint(),
-                    goalDefinition.shouldRollbackIfUnsatisfied()));
+            .map(coexistenceGoalTransform());
+  }
+
+  /**
+   * This convert is in a helper function in order to define the generic variables T1 and T2
+   */
+  private static <T1 extends TimingConstraint, T2 extends TimingConstraint>
+  Convert<
+      Pair<Pair<Pair<Pair<Pair<Pair<ActivityTemplate, Optional<ConstraintExpression.ActivityExpression>>, String>, ConstraintExpression>, Optional<T1>>, Optional<T2>>, Boolean>,
+      GoalSpecifier.CoexistenceGoalDefinition>
+  coexistenceGoalTransform() {
+    return Convert.between(untuple(GoalSpecifier.CoexistenceGoalDefinition::new), (GoalSpecifier.CoexistenceGoalDefinition $) -> tuple(
+        $.activityTemplate(),
+        $.activityFinder(),
+        $.alias,
+        $.forEach,
+        (Optional<T1>) $.startConstraint,
+        (Optional<T2>) $.endConstraint,
+        $.shouldRollbackIfUnsatisfied
+    ));
```

Comment:
Can you give me more details about this ? Why don't we use the usual mapping seen in other parsers in this class ?

Reply:
That's necessary to convert from VariadicProductParser to JsonObjectParser. After speaking to Matt (with whom I pair programmed the code of this PR) and making some further investigation, I don't see any other parser method that could do this same job.